### PR TITLE
Build CUDA image for lintrunner

### PR DESCRIPTION
**Pull request by cyyever**
Following the recent works, it is necessary to add CUDA files in the docker container so that we can lint CUDA code in the future.

cc @ezyang  @Skylion007  @malfet

### CODE REVIEW

File: .ci/docker/build.sh [hunk 1]
```diff
@@ -316,7 +324,6 @@ fi
 # Build image
 docker build \
        --no-cache \
-       --progress=plain \
```
Comment (malfet) [inline .ci/docker/build.sh:319]: Why are you changing this?

**Review by malfet — APPROVED** [review]
Looks good to me (especially as this is a no op at the moment), but please revert progress bar change or explain why it is needed)

File: .ci/docker/build.sh (hunk 1 above)
Comment (cyyever) [reply .ci/docker/build.sh:319]: Latest docker seems discarding the option, at least on docker 24.0.5. Actually, I saw the error message:
```
DEPRECATED: The legacy builder is deprecated and will be removed in a future release.
            Install the buildx component to build images with BuildKit:
            https://docs.docker.com/go/buildx/

unknown flag: --progress
```

Comment (cyyever) [reply .ci/docker/build.sh]: Reverted

**Comment by cyyever** [comment]
Of course I have no permission to push

**Pull request closed without merging**